File: 08-23/main.cpp
#include "adts/bst/bstImp.cpp"
#include <iostream>

int main(){
    int i;
    bst<int>* t = new bstImp<int>();

    do
    {
        std::cin >> i;
        t->insert(i);
    } while (i != 0);
    

    t->printInOrder();
    std::cout << "Quitar numero: ";
    std::cin >> i;
    t->remove(i);
    t->printInOrder();

    return 0;
}

File: 08-23/Main.java
import adts.bst.BST;
import adts.bst.BSTImp;

public class Main {
    public static void main(String[] args) {
        BST<Integer> bst = new BSTImp<Integer>();

        int lastInsert = 0;
        for(int i = 0; i < 5; i++){
            lastInsert = (int)(Math.random()*100);
            bst.insert(lastInsert);
        }

        bst.printInOrder();

        System.out.println("Remove " + lastInsert);
        bst.remove(lastInsert);
        bst.printInOrder();
    }
}
